blocks/api/factory.js

```diff
@@ -7,26 +7,30 @@ import { get, castArray, findIndex, isObjectLike, find } from 'lodash';
 /**
  * Internal dependencies
  */
-import { getBlockSettings } from './registration';
+import { getBlockType } from './registration';
 
 /**
- * Returns a block object given its type and attributes
+ * Returns a block object given its type and attributes.
  *
- * @param  {String} blockType   BlockType
+ * @param  {String} blockName   Block name
  * @param  {Object} attributes  Block attributes
  * @return {Object}             Block object
  */
-export function createBlock( blockType, attributes = {} ) {
-	const blockSettings = getBlockSettings( blockType );
+export function createBlock( blockName, attributes = {} ) {
```

Comment:
I think `blockName` should remain `blockType`.  Where you can interpret the value to be the ID of the blockType.